agenthub/langchains_agent/langchains_agent.py

```diff
@@ -58,21 +59,58 @@
 
 class LangchainsAgent(Agent):
     _initialized = False
+    _current_directory = "/"
 
     def __init__(self, llm: LLM):
         super().__init__(llm)
         self.monologue = Monologue()
         self.memory = LongTermMemory()
 
     def _add_event(self, event: dict):
-        if 'args' in event and 'output' in event['args'] and len(event['args']['output']) > MAX_OUTPUT_LENGTH:
-            event['args']['output'] = event['args']['output'][:MAX_OUTPUT_LENGTH] + "..."
+        if 'args' in event and 'output' in event['args']:
+            if len(event['args']['output']) > MAX_OUTPUT_LENGTH:
+                event['args']['output'] = event['args']['output'][:MAX_OUTPUT_LENGTH] + "..."
+            if event['action'] == "browse":
+                soup = BeautifulSoup(event['args']['output'], 'html.parser')
+                for non_text_tag in soup.find_all(["script", "style", "iframe", "noscript", "header", "footer", "nav"]):
+                    non_text_tag.decompose()
+                text = ' '.join(soup.stripped_strings)
+                event['args']['output'] = text
```

Comment:
Thi is great! Let's move it into the browse action itself, so other agents can take advantage of it